decidim-core/app/commands/decidim/messaging/reply_to_chat.rb

```diff
@@ -33,6 +33,22 @@ def call
 
       private
 
+      def sender
+        form.current_user
+      end
+
+      def message
+        @message ||= chat.add_message(sender: sender, body: form.body)
+      end
+
+      def notify_interlocutors
```

Comment:
Why not use the `Decidim::EventManager` here? Too verbose maybe?

Reply:
I'm not sure how this `EventManager` works, I checked how this type of emails are sent in other places and tried to do it similarly...

Reply:
Hi @deivid-rodriguez! Sorry to chime in so late. You can find an example of events working here:

https://github.com/decidim/decidim/blob/master/decidim-comments/app/commands/decidim/comments/create_comment.rb#L46

and here:

https://github.com/decidim/decidim/blob/master/decidim-comments/app/events/decidim/comments/comment_created_event.rb

Basically, an `Event` is an abstraction that creates a notification or sends an email behind the scenes, depending on which module is included (`Decidim::Events::EmailEvent` or `Decidim::Events::NotificationEvent`).

Do you think you could switch to use that? It's a really good use case for notifications :)

Reply:
Ok, will do!

Reply:
Thanks for the pointers, by the way. It makes sense to use that indeed.

Reply:
Ok, so I had a look at this and I'd rather leave it as it is because:

* There's no other example where these `Event`'s  don't create a notification but only send emails. Other places where only emails need to be sent (joining meetings, devise, report/moderation email alerts) use mailers directly.
* Current `EmailNotificationGenerator` does not support conditionally sending emails upon certain conditions on the recipients and the base resource, which I'm doing here.
* I'm not sure it would make the code better in this case. It introduces an indirection that feels like unnecesarily complicating things to me.

Reply:
After a second thought, I guess I can do the recipient filtering before publishing the event, so the second point wouldn't be a blocker. But I still think this introduces an unnecessary level of complexity for little gain. Anyways, let me know and I'll update this as you please.

Reply:
Well, the main idea here is to have a visual indication somewhere that you have unread messages, as well as receiving an email notification. But now that I give it a second thought, maybe the nature of that notification is different. 

Facebook, for example, uses two different icons in the top bar to indicate unread conversations and *events*. Maybe we should treat them differently and do something similar in `Decidim`?

We can of course do this in a further PR, but I think it's quite valuable to have this on the first iteration.

To be clear, I mean something like:

![image](https://user-images.githubusercontent.com/111746/33422258-619d203a-d5b5-11e7-92b1-d5de5e269f75.png)

Reply:
@josepjaume Note that I'm already doing that in this PR. There's a separate indicator for messages in the top menu that works just like the notification indicator: it gets hightlighted if there are new unread messages. Have a look at the screenshots!